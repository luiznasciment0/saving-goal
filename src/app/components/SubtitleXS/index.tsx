import Typography, { ColorProps } from 'app/components/Typography'

interface SubtitleXSProps {
  children: React.ReactNode
  color?: ColorProps
  tag?: 'h1' | 'h6'
}

const SubtitleXS = ({
  children,
  color = 'primaryBlue',
  tag = 'h1'
}: SubtitleXSProps) => (
  <Typography weight="light" size="large" color={color} as={tag}>
    {children}
  </Typography>
)

export default SubtitleXS
