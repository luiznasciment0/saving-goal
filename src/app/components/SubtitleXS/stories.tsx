import { Story, Meta } from '@storybook/react/types-6-0'
import SubtitleXS from '.'

export default {
  title: 'SubtitleXS',
  component: SubtitleXS,
  argTypes: {
    children: {
      type: 'string'
    }
  }
} as Meta

export const Default: Story = (args) => <SubtitleXS>{args.children}</SubtitleXS>

Default.args = {
  children: 'Test'
}
