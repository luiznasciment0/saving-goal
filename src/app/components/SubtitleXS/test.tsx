import { render, screen } from '@testing-library/react'

import SubtitleXS from '.'

describe('<SubtitleXS />', () => {
  it('should render the SubtitleXS and its styles', () => {
    const { container } = render(<SubtitleXS>Test</SubtitleXS>)

    expect(screen.getByRole('heading', { name: /test/i })).toHaveStyle({
      'font-size': '1.8rem',
      'line-height': '2.16rem',
      'font-weight': '400',
      color: '#1b31a8'
    })

    expect(container.firstChild).toMatchSnapshot()
  })
})
