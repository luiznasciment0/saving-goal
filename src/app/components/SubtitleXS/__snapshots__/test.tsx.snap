// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubtitleXS /> should render the SubtitleXS and its styles 1`] = `
.c0 {
  font-size: 1.8rem;
  line-height: 2.16rem;
  font-family: 'Work Sans';
  color: #1b31a8;
  font-weight: 400;
}

<h1
  class="c0"
  color="primaryBlue"
>
  Test
</h1>
`
